src/components/views/messages/TextualBody.js

```diff
@@ -59,10 +64,23 @@ module.exports = React.createClass({
 
             // track whether the preview widget is hidden
             widgetHidden: false,
+
+            editMode: false,
         };
     },
 
+    onAction(payload) {
+        // This isn't very nice -- but make sure it is the textualBody for the same
+        // mxEvent as the MessageContextMenu was generated for
+        // localize subscribe/dispatch?
+        if (payload.action === 'edit' && payload.mxEvent === this.props.mxEvent) {
+            this.state.editMode = !this.state.editMode
+            this.forceUpdate()
```

Comment:
yup, indeed, 99% of the time calling forceUpdate() is a mistake.
in react, you never ever directly mutate a component's state; instead you go via `this.setState({editMode: !this.state.editMode});`, which does the necessary magic for the component to know it has to re-render itself.